sdk/rust/src/hbar.rs

```diff
@@ -0,0 +1,268 @@
+use std::fmt::{
+    Debug,
+    Display,
+    Formatter,
+};
+use std::str::FromStr;
+
+use derive_more::{
+    Add,
+    AddAssign,
+    Div,
+    DivAssign,
+    Mul,
+    MulAssign,
+    Neg,
+    Sub,
+    SubAssign,
+};
+use rust_decimal::prelude::*;
+use serde_with::{
+    DeserializeFromStr,
+    SerializeDisplay,
+};
+
+use crate::Error;
+
+pub type Tinybar = i64;
+
+#[repr(i64)]
+#[derive(Debug, SerializeDisplay, Copy, DeserializeFromStr, Hash, PartialEq, Eq, Clone)]
+pub enum HbarUnit {
+    /// The atomic (smallest) unit of [`Hbar`], used natively by the Hedera network.
+    ///
+    /// It is equivalent to <sup>1</sup>&frasl;<sub>100,000,000</sub> [`Hbar`](Self::Hbar).
+    Tinybar = 1,
+
+    /// Equivalent to 100 [`Tinybar`](Self::Tinybar) or <sup>1</sup>&frasl;<sub>1,000,000</sub> [`Hbar`](Self::Hbar).
+    Microbar = 100,
+
+    /// Equivalent to 100,000 [`Tinybar`](Self::Tinybar) or <sup>1</sup>&frasl;<sub>1,000</sub> [`Hbar`](Self::Hbar).
+    Millibar = 100_000,
+
+    /// The base unit of [`Hbar`], equivalent to 100 million [`Tinybar`](Self::Tinybar).
+    Hbar = 100_000_000,
+
+    /// Equivalent to 1 thousand [`Hbar`](Self::Hbar) or 100 billion [`Tinybar`](Self::Tinybar).
+    Kilobar = 1_000 * 100_000_000,
+
+    /// Equivalent to 1 million [`Hbar`](Self::Hbar) or 100 trillion [`Tinybar`](Self::Tinybar).
+    Megabar = 1_000_000 * 100_000_000,
+
+    /// Equivalent to 1 billion [`Hbar`](Self::Hbar) or 100 quadrillion [`Tinybar`](Self::Tinybar).
+    ///
+    /// The maximum hbar amount supported by Hedera in any context is ~92 gigabar
+    /// (2<sup>63</sup> tinybar); use this unit sparingly.
+    Gigabar = 1_000_000_000 * 100_000_000,
+}
+
+impl HbarUnit {
+    #[must_use]
+    pub const fn tinybars(self) -> Tinybar {
+        self as Tinybar
+    }
+
+    #[must_use]
+    pub const fn symbol(self) -> &'static str {
+        match self {
+            HbarUnit::Tinybar => "tℏ",
+            HbarUnit::Microbar => "μℏ",
+            HbarUnit::Millibar => "mℏ",
+            HbarUnit::Hbar => "ℏ",
+            HbarUnit::Kilobar => "kℏ",
+            HbarUnit::Megabar => "Mℏ",
+            HbarUnit::Gigabar => "Gℏ",
+        }
+    }
+}
+
+impl Display for HbarUnit {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.symbol())
+    }
+}
+
+impl FromStr for HbarUnit {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "tℏ" => Ok(HbarUnit::Tinybar),
+            "μℏ" => Ok(HbarUnit::Microbar),
+            "mℏ" => Ok(HbarUnit::Millibar),
+            "ℏ" => Ok(HbarUnit::Hbar),
+            "kℏ" => Ok(HbarUnit::Kilobar),
+            "Mℏ" => Ok(HbarUnit::Megabar),
+            "Gℏ" => Ok(HbarUnit::Gigabar),
+            _ => Err(Error::basic_parse(format!(
+                "Given string `{s}` was not recognized as an Hbar unit symbol"
+            ))),
+        }
+    }
+}
+
+#[derive(
+    SerializeDisplay,
```

Comment:
@Sean-Tedrow-LB We need to serialize/deserialize as a numerical _tinybar_ value.